packages/react-scripts/scripts/utils/createJestConfig.js

```diff
@@ -11,18 +11,20 @@ const fs = require('fs');
 const chalk = require('chalk');
 const paths = require('../../config/paths');
 
+const useTypeScript = fs.existsSync(paths.appTsConfig);
+
 module.exports = (resolve, rootDir, isEjecting) => {
   // Use this instead of `paths.testsSetup` to avoid putting
   // an absolute filename into configuration after ejecting.
   const setupTestsMatches = paths.testsSetup.match(/src\/setupTests\.(.+)/);
   const setupTestsFileExtension =
-    (setupTestsMatches && setupTestsMatches[1]) || 'js';
+    (setupTestsMatches && setupTestsMatches[1]) ||
+    (useTypeScript ? 'ts' : 'js');
```

Comment:
I'm not sure this is the best way to fix this. Why doesn't the regex just work like it should?